pkg/splunk/enterprise/searchheadcluster.go

```diff
@@ -147,12 +148,17 @@ func ApplySearchHeadCluster(client splcommon.ControllerClient, cr *enterpriseApi
 	}
 	cr.Status.Phase = phase
 
-	// no need to requeue if everything is ready
-	if cr.Status.Phase == splcommon.PhaseReady {
-		if cr.Status.AppContext.AppsSrcDeployStatus != nil {
-			markAppsStatusToComplete(client, cr, &cr.Spec.AppFrameworkConfig, cr.Status.AppContext.AppsSrcDeployStatus)
+	if cr.Status.AppContext.AppsSrcDeployStatus != nil && cr.Status.DeployerPhase == splcommon.PhaseReady {
```

Comment:
Does this mean for SHC members ready state is not based on the app install status?

Reply:
Correct. Deployer will not come to Phase ready unless the SHC is up and apps are really installed.